src/configuration.py

```diff
@@ -0,0 +1,165 @@
+DUNE_KEY = os.getenv("DUNE_KEY")
+ETHERSCAN_KEY = os.getenv("ETHERSCAN_KEY")
+
+ADDRESS = "0x9008D19f58AAbD9eD0D60971565AA8510560ab41"
+
+def get_solver_dict() -> Dict[str, List[int]]:
+    """
+    Function prepares a solver dictionary by fetching solver names from a Dune query.
+    """
+    solver_dict = {}
+    query = Query(
+        name="Solver Dictionary",
+        query_id=1372857,
+    )
+    if DUNE_KEY is not None:
+        dune = DuneClient(DUNE_KEY)
+        results = dune.refresh(query)
+        solvers = results.get_rows()
+        for solver in solvers:
+            solver_dict[solver["name"]] = [0, 0]
+
+        # These names need to be updated since Dune and Orderbook Endpoint have different names.
+        # Example, "1Inch: [0, 0]" is a specific row, the first value is the number of solutions
+        # won, second value is number of solutions of that solver with higher surplus found.
+
+        solver_dict["BaselineSolver"] = solver_dict.pop("Baseline")
+        solver_dict["1Inch"] = solver_dict.pop("Gnosis_1inch")
+        solver_dict["0x"] = solver_dict.pop("Gnosis_0x")
+        solver_dict["BalancerSOR"] = solver_dict.pop("Gnosis_BalancerSOR")
+        solver_dict["ParaSwap"] = solver_dict.pop("Gnosis_ParaSwap")
+        solver_dict["SeaSolver"] = solver_dict.pop("Seasolver")
+        solver_dict["CowDexAg"] = solver_dict.pop("DexCowAgg")
+        solver_dict["NaiveSolver"] = solver_dict.pop("Naive")
+
+    return solver_dict
+
+
+def get_logger(filename: Optional[str] = None) -> logging.Logger:
+    """
+    get_logger() returns a logger object that can write to a file, terminal or only file if needed.
+    """
+    logging.basicConfig(format="%(levelname)s - %(message)s")
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    if filename:
+        file_handler = logging.FileHandler(filename + ".log", mode="w")
+        file_handler.setLevel(logging.INFO)
+        formatter = logging.Formatter("%(levelname)s - %(message)s")
+        file_handler.setFormatter(formatter)
+        logger.addHandler(file_handler)
+
+    return logger
+
+
+def get_tx_hashes_by_block(web_3, start_block: int, end_block: int) -> List[str]:
+    """
+    Function filters hashes by contract address, and block ranges
+    """
+    filter_criteria = {
+        "fromBlock": int(start_block),
+        "toBlock": int(end_block),
+        "address": ADDRESS,
+    }
+    # transactions may have repeating hashes, since even event logs are filtered
+    # therefore, check if hash has already been added to the list
+
+    # only successful transactions are filtered
+    transactions = web_3.eth.filter(filter_criteria).get_all_entries()
+    settlement_hashes_list = []
+    for transaction in transactions:
+        tx_hash = (transaction["transactionHash"]).hex()
+        if tx_hash not in settlement_hashes_list:
+            settlement_hashes_list.append(tx_hash)
+    return settlement_hashes_list
+
+
+
+def get_eth_value():
```

Comment:
So, this function ideally should only be responsible for getting the eth/usdc price. I would remove the rest. The min_flag_usd seems like a global constant, and the eth_absolute_check should be computed at the moment we are about to do the checks I guess